backend/tests/projects/toolmodels/conftest.py

```diff
@@ -172,6 +172,113 @@ def fixture_mock_git_rest_api_for_artifacts(
             yield responses
 
 
+@pytest.fixture(name="mock_git_get_commit_information_api")
+def fixture_mock_git_get_commit_information_api(
+    request: pytest.FixtureRequest, git_type: git_models.GitType
+):
+    match git_type:
+        case git_models.GitType.GITLAB:
+            responses.get(
+                "https://example.com/api/v4/projects/10000/repository/commits?ref_name=main&path=model-complexity-badge.svg",
+                status=200,
+                json=[
+                    {
+                        "id": "ee85bc253111b7a8ca2ce5aa26b8f5f36325f48a",
+                        "created_at": "2050-04-11T10:09:59.000+02:00",
+                        "title": "test: Test commit message",
+                        "message": "test: Test commit message\n",
+                        "author_name": "test-name",
+                        "author_email": "test-email",
+                        "authored_date": "2050-04-11T10:09:59.000+02:00",
+                        "committer_name": "test-name",
+                        "committer_email": "test-email",
+                        "committed_date": "2050-04-11T10:09:59.000+02:00",
+                    }
+                ],
+            )
+            responses.get(
+                "https://example.com/api/v4/projects/10000/repository/commits?ref_name=diagram-cache/main&path=diagram_cache/_c90e4Hdf2d2UosmJBo0GTw.svg",
```

Comment:
Instead of duplicating the same call three times, you can use a regex expression to match all request calls with a specific wildcard: https://github.com/getsentry/responses#dynamic-responses